pkg/repo/index_test.go

```diff
@@ -214,6 +214,81 @@ func TestMerge(t *testing.T) {
 
 }
 
+func TestMergePrerelease(t *testing.T) {
+	ind1 := NewIndexFile()
+	ind1.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.2.0",
+	}, "dreadnought-0.2.0.tgz", "http://example.com", "aaaa")
+
+	ind2 := NewIndexFile()
+	ind2.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.1.0",
+	}, "dreadnought-0.1.0.tgz", "http://example.com", "aaaabbbb")
+	ind2.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.1-dev",
```

Comment:
This is an interesting case here... This tag does not comply to the [Semantic Versioning 2.0.0 specification](https://semver.org/spec/v2.0.0.html). Previously these versions would be rejected to protect us from potential connection string injection attacks.

The old behaviour was to reject this version from being included to protect from those attacks. I wonder if `0.1.0-dev` would work with the old code instead.

Reply:
Good catch, it doesn't match semver v2. That's worth exploring whether my original use case is resolved based on that.

> Previously these versions would be rejected

IIRC, this was not the behaviour I observed. It was more like "Depending on whether you create an index with the `v0.1-dev` style first and then merge in other indexes like `v0.1`, or if you first create the index with `v0.1` and then try to merge `v0.1-dev`, the result is different."

Reply:
I took a brief look again and indeed, if we simply use `0.1.0-dev` style dev versions then Helm does the right thing.

If however we use `0.1-dev` style dev versions, then depending on the merge ordering, Helm will inconsistently either keep or discard the `0.1-dev` versions from the index.

There is no indication to the user that the `v0.1-dev` version is invalid, it's just silently discarded in some cases depending on how you invoke helm.